Python/Product/Analysis/LanguageServer/Structures.cs

```diff
@@ -479,45 +515,61 @@ public struct RenameCapabilities { public bool dynamicRegistration; }
     /// client capabilities following the specification for extra settings.
     /// </summary>
     [Serializable]
-    public class PythonClientCapabilities {
+    public struct PythonClientCapabilities {
+        [JsonConstructor]
+        public PythonClientCapabilities([DefaultValue(false)] bool analysisUpdates,
+            [DefaultValue(Timeout.Infinite)] int completionsTimeout,
+            [DefaultValue(false)] bool traceLogging,
+            [DefaultValue(false)] bool liveLinting) {
+            AnalysisUpdates = analysisUpdates;
+            CompletionsTimeout = completionsTimeout;
+            TraceLogging = traceLogging;
+            LiveLinting = liveLinting;
+        }
+
         /// <summary>
         /// Client expects analysis progress updates, including notifications
         /// when analysis is complete for a particular document version.
         /// </summary>
-        public bool? analysisUpdates;
+        public readonly bool AnalysisUpdates;
```

Comment:
Why change these from how the JSON is represented?

Reply:
Because inside C# code we don't care about JSON (that's the whole point of these types) and everywhere else public fields and properties are UpperCamelCase (which is also default for C#). Names in constructor match names in JSON

Reply:
As the author of these types, I say the whole point is to represent the JSON messages as closely as possible :) (See my comment below for my actual argument about not changing this)